fast64_internal/f3d/f3d_material.py

```diff
@@ -177,85 +177,144 @@ def update_draw_layer(self, context):
+    return (
+        settings.blend_p1 == color or settings.blend_m1 == color
+        or (is_one_cycle and (settings.blend_p2 == color or settings.blend_m2 == color))
+    )
+    
 
-    is_one_cycle = f3dMat.rdp_settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+def does_blender_use_alpha(
+    settings: "RDPSettings",
+    alpha: str,
+    default_for_no_rendermode: bool = False
+) -> bool:
+    if not settings.set_rendermode:
+        return default_for_no_rendermode
+    is_one_cycle = settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+    return settings.blend_a1 == alpha or (is_one_cycle and settings.blend_a2 == alpha)
 
-    if f3dMat.rdp_settings.set_rendermode:
-        if f3dMat.rdp_settings.rendermode_advanced_enabled:
-            if f3dMat.rdp_settings.cvg_x_alpha:
-                blend_method = "CLIP"
-            elif (
-                is_one_cycle
-                and f3dMat.rdp_settings.force_bl
-                and f3dMat.rdp_settings.blend_p1 == "G_BL_CLR_IN"
-                and f3dMat.rdp_settings.blend_a1 == "G_BL_A_IN"
-                and f3dMat.rdp_settings.blend_m1 == "G_BL_CLR_MEM"
-                and f3dMat.rdp_settings.blend_b1 == "G_BL_1MA"
-            ):
-                blend_method = "BLEND"
-            elif (
-                not is_one_cycle
-                and f3dMat.rdp_settings.force_bl
-                and f3dMat.rdp_settings.blend_p2 == "G_BL_CLR_IN"
-                and f3dMat.rdp_settings.blend_a2 == "G_BL_A_IN"
-                and f3dMat.rdp_settings.blend_m2 == "G_BL_CLR_MEM"
-                and f3dMat.rdp_settings.blend_b2 == "G_BL_1MA"
-            ):
-                blend_method = "BLEND"
-            else:
-                blend_method = "OPAQUE"
-        else:
-            rendermode = f3dMat.rdp_settings.rendermode_preset_cycle_1
-            if not is_one_cycle:
-                rendermode = f3dMat.rdp_settings.rendermode_preset_cycle_2
-
-            f3d = get_F3D_GBI()
-            r_mode = getattr(f3d, rendermode, f3d.G_RM_AA_ZB_OPA_SURF)
-            if r_mode & f3d.CVG_X_ALPHA:
-                blend_method = "CLIP"
-            else:
-                cfunc = GBL_c1 if is_one_cycle else GBL_c2
-                xlu_comb = r_mode & cfunc(f3d.G_BL_CLR_IN, f3d.G_BL_A_IN, f3d.G_BL_CLR_MEM, f3d.G_BL_1MA)
-                if xlu_comb and r_mode & f3d.FORCE_BL:
-                    blend_method = "BLEND"
-                else:
-                    blend_method = "OPAQUE"
 
-    return blend_method
+def does_blender_use_mix(
+    settings: "RDPSettings",
+    mix: str,
+    default_for_no_rendermode: bool = False
+) -> bool:
+    if not settings.set_rendermode:
+        return default_for_no_rendermode
+    is_one_cycle = settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+    return settings.blend_b1 == mix or (is_one_cycle and settings.blend_b2 == mix)
+
+
+def is_blender_lerp(
+    settings: "RDPSettings",
+    cycle: int,
+    p: str,
+    a: str,
+    m: str,
+    b: str,
```

Comment:
idk a lot about this but what's the meaning of those letters here? if with the context it makes sense then ignore this, otherwise I'd like to have comments next to those parameters to say what it is

Reply:
these are the arguments used for the blending function. They're referred to with these letter designations in the programming manual

Reply:
Correct, these are their names. P and M probably stand for plus and minus, but A and B are just arbitrary letters.